src/rational/mat_q/arithmetic/mul_scalar.rs

```diff
@@ -0,0 +1,301 @@
+
+use super::super::MatQ;
+use crate::integer::Z;
+use crate::macros::arithmetics::{
+    arithmetic_trait_borrowed_to_owned, arithmetic_trait_mixed_borrowed_owned,
+    arithmetic_trait_reverse,
+};
+use crate::macros::for_others::implement_for_others;
+use crate::rational::Q;
+use crate::traits::{GetNumColumns, GetNumRows};
+use flint_sys::fmpq_mat::{fmpq_mat_scalar_mul_fmpq, fmpq_mat_scalar_mul_fmpz};
+use std::ops::Mul;
+
+impl Mul<&Z> for &MatQ {
+    type Output = MatQ;
+    /// Implements multiplication for a [`MatQ`] matrix with a [`Z`] integer.
+    ///
+    /// Parameters:
+    /// - `scalar`: specifies the scalar by which the matrix is multiplied
+    ///
+    /// Returns the product of `self` and `other` as a [`MatQ`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    /// use qfall_math::integer::Z;
+    /// use std::str::FromStr;
+    ///
+    /// let mat1 = MatQ::from_str("[[2/3,1],[1/2,2]]").unwrap();
+    /// let integer = Z::from(3);
+    ///
+    /// let mat2 = &mat1 * &integer;
+    /// ```
+    fn mul(self, scalar: &Z) -> Self::Output {
+        let mut out = MatQ::new(self.get_num_rows(), self.get_num_columns()).unwrap();
+        unsafe {
+            fmpq_mat_scalar_mul_fmpz(&mut out.matrix, &self.matrix, &scalar.value);
+        }
+        out
+    }
+}
+
+arithmetic_trait_reverse!(Mul, mul, Q, MatQ, MatQ);
+
+arithmetic_trait_borrowed_to_owned!(Mul, mul, MatQ, Q, MatQ);
+arithmetic_trait_borrowed_to_owned!(Mul, mul, Q, MatQ, MatQ);
+arithmetic_trait_mixed_borrowed_owned!(Mul, mul, MatQ, Q, MatQ);
+arithmetic_trait_mixed_borrowed_owned!(Mul, mul, Q, MatQ, MatQ);
+
+implement_for_others!(Z, MatQ, Mul Scalar for i8 i16 i32 i64 u8 u16 u32 u64);
+
+impl Mul<&Q> for &MatQ {
+    type Output = MatQ;
+    /// Implements multiplication for a [`MatQ`] matrix with a [`Q`] rational.
+    ///
+    /// Parameters:
+    /// - `scalar`: specifies the scalar by which the matrix is multiplied
+    ///
+    /// Returns the product of `self` and `other` as a [`MatQ`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    /// use qfall_math::rational::Q;
+    /// use std::str::FromStr;
+    ///
+    /// let mat1 = MatQ::from_str("[[2/3,1],[1/2,2]]").unwrap();
+    /// let rational = Q::from(3/7);
+    ///
+    /// let mat2 = &mat1 * &rational;
+    /// ```
+    fn mul(self, scalar: &Q) -> Self::Output {
+        let mut out = MatQ::new(self.get_num_rows(), self.get_num_columns()).unwrap();
+        unsafe {
+            fmpq_mat_scalar_mul_fmpq(&mut out.matrix, &self.matrix, &scalar.value);
+        }
+        out
+    }
+}
+
+arithmetic_trait_reverse!(Mul, mul, Z, MatQ, MatQ);
+
+arithmetic_trait_borrowed_to_owned!(Mul, mul, MatQ, Z, MatQ);
+arithmetic_trait_borrowed_to_owned!(Mul, mul, Z, MatQ, MatQ);
+arithmetic_trait_mixed_borrowed_owned!(Mul, mul, MatQ, Z, MatQ);
+arithmetic_trait_mixed_borrowed_owned!(Mul, mul, Z, MatQ, MatQ);
+
+#[cfg(test)]
+mod test_mul_z {
```

Comment:
Please add a test for `availability`, i.e. a test that checks (almost) any input types that should work for this function.
You've tested a few, but if one of them shouldn't work due to changes of the implementation anymore, it should be easily detectable.